cmd/kubeadm/app/cmd/phases/markmaster.go

```diff
@@ -25,68 +25,65 @@ import (
-	kubeadmscheme.Scheme.Default(cfg)
-
-	var cfgPath string
-	kubeConfigFile := kubeadmconstants.GetAdminKubeConfigPath()
 
-	cmd := &cobra.Command{
-		Use:     "mark-master",
-		Short:   "Mark a node as master",
-		Long:    markMasterLongDesc,
-		Example: markMasterExample,
-		Aliases: []string{"markmaster"},
-		Run: func(cmd *cobra.Command, args []string) {
-			if err := validation.ValidateMixedArguments(cmd.Flags()); err != nil {
-				kubeadmutil.CheckErr(err)
-			}
-
-			kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
-			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
-			kubeadmutil.CheckErr(err)
+type markcontrolplaneData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+}
 
-			// KubernetesVersion is not used, but we set it explicitly to avoid the lookup
-			// of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
-			SetKubernetesVersion(cfg)
+// NewmarkcontrolplanePhase creates a kubeadm workflow phase that implements mark-controlplane checks.
+func NewmarkcontrolplanePhase() workflow.Phase {
+	return workflow.Phase{
+		Name:    "markcontrolplane",
+		Short:   "Run master markcontrolplane checks",
+		Long:    "Run master markcontrolplane checks, applies master label.",
+		Example: markcontrolplaneExample,
+		Run:     runmarkcontrolplane,
+	}
+}
 
-			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, cfg)
-			kubeadmutil.CheckErr(err)
+// runmarkcontrolplane executes markcontrolplane checks logic.
+func runmarkcontrolplane(c workflow.RunData) error {
+	data, ok := c.(markcontrolplaneData)
+	if !ok {
+		return fmt.Errorf("markcontrolplane phase invoked with an invalid data struct")
+	}
 
-			err = markmasterphase.MarkMaster(client, internalcfg.NodeRegistration.Name, internalcfg.NodeRegistration.Taints)
-			kubeadmutil.CheckErr(err)
-		},
+	fmt.Println("[markcontrolplane] running markcontrolplane checks")
+	var cfgPath string
+    kubeConfigFile := kubeadmconstants.GetAdminKubeConfigPath()
+	kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
+	client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
+	if err != nil {
+		return err
+	}
+	internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, data.Cfg())
+	if err != nil {
+		return err
 	}
 
-	options.AddKubeConfigFlag(cmd.Flags(), &kubeConfigFile)
-	cmd.Flags().StringVar(&cfgPath, "config", cfgPath, "Path to kubeadm config file. WARNING: Usage of a configuration file is experimental")
-	cmd.Flags().StringVar(&cfg.NodeRegistration.Name, "node-name", cfg.NodeRegistration.Name, `The node name to which label and taints should apply`)
+	if err := markmasterphase.MarkMaster(client, internalcfg.NodeRegistration.Name, internalcfg.NodeRegistration.Taints); err != nil {
```

Comment:
i think it's ok to rename the MarkMaster function to MarkControlPlane as part of this refector.
basically `master->controlPlane` everywhere.